Examples/src/components/Examples/Charts2D/BasicChartTypes/RealTimeMountainChart/index.tsx

```diff
@@ -0,0 +1,107 @@
+        y1: initialValues.yValues[ initialValues.yValues.length - 1 ],
+        xCoordShift: -20,
+        yCoordShift: -20,
+        svgString:
+            '<svg width="40" height="40" xmlns="http://www.w3.org/2000/svg"><circle cx="20" cy="20" fill="steelblue" r="1" stroke="steelblue"><animate attributeName="r" from="0" to="20" dur="1s" begin="0s" repeatCount="indefinite"/><animate attributeName="opacity" from="1" to="0" dur="1s" begin="0s" repeatCount="indefinite"/></circle><circle cx="20" cy="20" fill="steelblue" r="5"/></svg>'
+    });
+
+    sciChartSurface.annotations.add( pulsingDotAnnotation );
+
+    // This function performs animation on any XyDataSeries, animating the latest point only
+    // Be careful of reentrancy, e.g. calling animateXy more than once before previous animation has finished
+    // might require special handling
+    const animateXy = (xyDataSeries: XyDataSeries, endX: number, endY: number, duration: number, easing: TEasing) => {
+        const count = xyDataSeries.count();
+        const startX = xyDataSeries.getNativeXValues().get(count - 1);
+        const startY = xyDataSeries.getNativeYValues().get(count - 1);
+
+        // use the DoubleAnimator class in scichart/Core/Animations/ to setup an animation from 0...1
+        DoubleAnimator.animate(0, 1, duration, (interpolationFactor) => {
+                // Using the interpolation factor (ranges from 0..1) compute the X,Y value now
+                const currentX = (endX - startX) * interpolationFactor + startX;
+                const currentY  = (endY - startY) * interpolationFactor + startY;
+
+                // Update X,Y value by direct access to the inner webassembly arrays
+                xyDataSeries.getNativeXValues().set(count - 1, currentX);
+                xyDataSeries.getNativeYValues().set(count - 1, currentY);
+
+                // update location of pulsing dot
+                pulsingDotAnnotation.x1 = currentX;
+                pulsingDotAnnotation.y1 = currentY;
+
+                // Force redraw
+                // can use xyDataSeries.notifyDataChanged(); to just update, but if we want to zoom to fit, we must use zoomExtents
+                sciChartSurface.zoomExtents();
+            },
+            () => {
+                // Animation complete, append the point
+                xyDataSeries.append(endX, endY);
+            }, easing);
+    };
+
+    // This is the loop where we add a new X,Y point and animate every 1 second to demonstrate animations
+    const addData = () => {
+        const generated = generator.getRandomWalkSeries(1);
+        const x = generated.xValues[0];
+        const y = generated.yValues[0];
+        animateXy(dataSeries, x, y, 250, easing.outExpo);
+        setTimeout(addData, 1000);
+    };
+
+    addData();
+    return sciChartSurface;
+};
+
+export default function RealtimeMountainChart() {
+    React.useEffect(() => {
+        drawExample();
```

Comment:
we should delete SciChartSurface, otherwise we get the memory leak. The correct way to delete is here - https://github.com/ABTSoftware/SciChart.JS.Examples/blob/master/Examples/src/components/Examples/Charts2D/BasicChartTypes/DigitalBandSeriesChart/index.tsx#L62

Reply:
fixed